cmd/modern/root/install/mssql-base.go

```diff
@@ -462,31 +468,51 @@ CHECK_POLICY=OFF`
 	}
 }
 
+func getDbNameIfExists(usingDbUrl string) string {
+	if strings.Contains(usingDbUrl, ",") {
+		dbNameStartIdx := strings.Index(usingDbUrl, ",") + 1
+		if dbNameStartIdx < len(usingDbUrl) {
+			dbName := usingDbUrl[dbNameStartIdx:]
+			dbName = strings.ReplaceAll(dbName, "'", "''")
+			dbName = strings.ReplaceAll(dbName, "]", "]]")
+			return dbName
+		}
+	}
+	return ""
+}
+
 func (c *MssqlBase) downloadAndRestoreDb(
 	controller *container.Controller,
 	containerId string,
 	userName string,
 ) {
 	output := c.Cmd.Output()
+	databaseName := getDbNameIfExists(c.usingDatabaseUrl)
+	databaseUrl := c.usingDatabaseUrl
+	if databaseName != "" {
+		databaseUrl = strings.Split(c.usingDatabaseUrl, ",")[0]
+	}
 
-	u, err := url.Parse(c.usingDatabaseUrl)
+	u, err := url.Parse(databaseUrl)
 	c.CheckErr(err)
-	_, file := filepath.Split(c.usingDatabaseUrl)
+	_, file := filepath.Split(databaseUrl)
 	fileNameWithNoExt := strings.TrimSuffix(file, filepath.Ext(file))
-
+	if databaseName == "" {
+		databaseName = fileNameWithNoExt
```

Comment:
>fileNameWithNoEx [](http://example.com/codeflow?start=17&length=16)

can the file name portion of a URL also have ] characters in it?

Reply:
Yes. It is possible. (At least on windows platform)

Reply:
Have escaped DbName generated from file name portion as well

Reply:
> > fileNameWithNoEx
> 
> can the file name portion of a URL also have ] characters in it?

I was wrong about this. The RFC for URL states that a filename can only contain `-`,`.`,`_`,`~`
So a filename in URL can never have `]`. The filename, if it contains the `]` character then it's most likely to be in its encoded form.
[RFC3986](https://www.rfc-editor.org/rfc/rfc3986#section-2.3)

Reply:
sure, which is why we should be decoding the Url properly not just scanning it for a comma